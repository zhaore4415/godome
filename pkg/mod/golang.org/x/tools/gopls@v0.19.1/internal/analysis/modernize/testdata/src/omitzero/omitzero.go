package omitzero

type Foo struct {
	EmptyStruct struct{} `json:",omitempty"` // want "Omitempty has no effect on nested struct fields"
}

type Bar struct {
	NonEmptyStruct struct{ a int } `json:",omitempty"` // want "Omitempty has no effect on nested struct fields"
}

type C struct {
	D string `json:",omitempty"`
}

type R struct {
	M string `json:",omitempty"`
}

type A struct {
	C C `json:"test,omitempty"` // want "Omitempty has no effect on nested struct fields"
	R R `json:"test"`
}

type X struct {
	NonEmptyStruct struct{ a int } `json:",omitempty" yaml:",omitempty"` // want "Omitempty has no effect on nested struct fields"
}

type Y struct {
	NonEmptyStruct struct{ a int } `yaml:",omitempty" json:",omitempty"` // want "Omitempty has no effect on nested struct fields"
}
