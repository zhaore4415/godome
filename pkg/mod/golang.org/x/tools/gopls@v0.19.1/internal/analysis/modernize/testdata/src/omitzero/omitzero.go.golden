-- Replace omitempty with omitzero (behavior change) --
package omitzero

type Foo struct {
	EmptyStruct struct{} `json:",omitzero"` // want "Omitempty has no effect on nested struct fields"
}

type Bar struct {
	NonEmptyStruct struct{ a int } `json:",omitzero"` // want "Omitempty has no effect on nested struct fields"
}

type C struct {
	D string `json:",omitempty"`
}

type R struct {
	M string `json:",omitempty"`
}

type A struct {
	C C `json:"test,omitzero"` // want "Omitempty has no effect on nested struct fields"
	R R `json:"test"`
}

type X struct {
	NonEmptyStruct struct{ a int } `json:",omitzero" yaml:",omitempty"` // want "Omitempty has no effect on nested struct fields"
}

type Y struct {
	NonEmptyStruct struct{ a int } `yaml:",omitempty" json:",omitzero"` // want "Omitempty has no effect on nested struct fields"
}

-- Remove redundant omitempty tag --
package omitzero

type Foo struct {
	EmptyStruct struct{} // want "Omitempty has no effect on nested struct fields"
}

type Bar struct {
	NonEmptyStruct struct{ a int } // want "Omitempty has no effect on nested struct fields"
}

type C struct {
	D string `json:",omitempty"`
}

type R struct {
	M string `json:",omitempty"`
}

type A struct {
	C C `json:"test"` // want "Omitempty has no effect on nested struct fields"
	R R `json:"test"`
}

type X struct {
	NonEmptyStruct struct{ a int } `yaml:",omitempty"` // want "Omitempty has no effect on nested struct fields"
}

type Y struct {
	NonEmptyStruct struct{ a int } `yaml:",omitempty"` // want "Omitempty has no effect on nested struct fields"
}
